library/ssl_tls.c

```diff
@@ -2693,6 +2693,21 @@ static int ssl_prepare_handshake_step( mbedtls_ssl_context *ssl )
 {
     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
 
+    /*
+     * We may have not been able to send to the peer all the handshake data
+     * that were written into the output buffer by the previous handshake step:
+     * the write to the network callback returned with the
+     * #MBEDTLS_ERR_SSL_WANT_WRITE error code.
```

Comment:
```suggestion
     * We may have not been able to send to the peer all the handshake data
     * that were written into the output buffer by the previous handshake step, if
     * the write to the network callback returned with the
     * #MBEDTLS_ERR_SSL_WANT_WRITE error code.
```
Nitpick, but I was struggling slightly to understand this comment.